gpii/node_modules/settingsHandlers/src/WebSocketsSettingsHandler.js

```diff
@@ -0,0 +1,154 @@
+
+    gpii.settingsHandlers.WebSocketsSettingsHandler.applySettingsForId = function (settings, id) {
+        var fallbackBlock = {settings: {}};
+
+        if (!(gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id])) {
+            gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id] = fallbackBlock;
+        }
+
+        var currentSettings = gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id].settings;
+
+        var applier = fluid.makeChangeApplier(currentSettings, {
+            resolverGetConfig: fluid.model.escapedGetConfig,
+            resolverSetConfig: fluid.model.escapedSetConfig
+        });
+
+        var newSettingsResponse = {};
+
+        fluid.each(settings, function (settingVal, settingKey) {
+            var oldValue = fluid.get(currentSettings, settingKey,
+                fluid.model.escapedGetConfig);
+
+            var change = {
+                path: settingKey,
+                value: settingVal
+            };
+            change.type = settingVal === undefined ? "DELETE" : "ADD";
+            applier.fireChangeRequest(change);
+
+            newSettingsResponse[settingKey] = {
+                "oldValue": oldValue,
+                "newValue": settingVal
+            };
+
+            fluid.set(currentSettings, settingKey, settings[settingKey],
+                      fluid.model.escapedSetConfig);
+        });
+
+        gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id].settings = currentSettings;
+
+        return newSettingsResponse;
+    };
+
+    gpii.settingsHandlers.WebSocketsSettingsHandler.setSettings = function (payload, id) {
+        var newSettingsResponse = [];
+
+        for (var i=0; i< payload.length; i++) {
+            newSettingsResponse[i] = {settings: gpii.settingsHandlers.WebSocketsSettingsHandler.applySettingsForId(payload[i].settings, id)};
+        }
+
+        gpii.settingsHandlers.WebSocketsSettingsHandler.notifySettings(id);
+
+        if (Object.keys(gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id].settings).length === 0) {
+            delete gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id];
+        }
+
+        return newSettingsResponse;
+    };
+
+    gpii.settingsHandlers.WebSocketsSettingsHandler.set = function (payload) {
+        var result = fluid.copy(payload);
+        for (var appId in payload) {
+            result[appId] = gpii.settingsHandlers.WebSocketsSettingsHandler.setSettings(payload[appId], appId);
+        }
+        return result;
+    };
+
+    gpii.settingsHandlers.WebSocketsSettingsHandler.addBrowser = function (id, socket) {
+        fluid.set(gpii.settingsHandlers.WebSocketsSettingsHandler.clients, [id, socket.id], socket);
+    };
+
+    gpii.settingsHandlers.WebSocketsSettingsHandler.removeBrowser = function (socket) {
+        for (var id in gpii.settingsHandlers.WebSocketsSettingsHandler.clients) {
+            if (socket.id in gpii.settingsHandlers.WebSocketsSettingsHandler.clients[id]) {
```

Comment:
This if branch is can be omitted - "delete" is guaranteed to be safe on a nonexistent property